wr-cli/parser.py

```diff
@@ -0,0 +1,103 @@
+from enum import Enum
+
+
+class ParserError(Enum):
+    EMPTY_STRING = 0
+    KEY_VALUE_PAIR = 1
+
+
+class Parser:
+    def _separate_tokens(self, user_input):
+        """
+        Helper to separate parts of the string according to some rules.
+
+        Parameters
+        ----------
+        user_input : string
+            input string
+
+        Returns
+        -------
+        list
+            array of separated string tokens
+        """
+        arr_user_string = user_input.split()
+        count = 0
+        stack = []
+        final_arr_user_string = []
+        for token in arr_user_string:
+            # keep the original token for comparisons, modified token for
+            # inserting into stack
+            modified_token = token.replace("\"", "")
+
+
+            if len(stack) != 0 and "\"" in token:
+                stack.append(modified_token)
+
+                final_token = " ".join(stack)
+                final_arr_user_string.append(final_token)
+
+                stack.clear()
+
+            elif len(stack) != 0 or ("\'" in token and token[1] != "\'"):
+                modified_token = token.replace("\'", "")
+                count +=1
+                if (count > 2):
+                    stack.append(modified_token)
+                else:
+                    final_arr_user_string.append(modified_token)
```

Comment:
This should be generalized to handle both double, single quotes, or whatever syntax we decide to use in the future. 

Per my suggestion in the main thread, I would change `count` to `wrap_character` where instead of keeping the count of the wrapping character (i.e. `'` or `"`) we keep track of the opening wrapping character itself.